plugins/web-terminal/src/components/TerminalComponent/TerminalComponent.tsx

```diff
@@ -1,27 +1,108 @@
-import { makeStyles } from '@material-ui/core';
-import React from 'react';
+import { Button, makeStyles, TextField } from '@material-ui/core';
+import React, { useCallback } from 'react';
 import { Terminal } from 'xterm';
 import { FitAddon } from 'xterm-addon-fit';
+import { AttachAddon } from 'xterm-addon-attach';
 import './static/xterm.css';
+import { InfoCard, Progress } from '@backstage/core-components';
 
 const useTerminalStyles = makeStyles({
   term: {
     width: '100%',
-    height: '100%',
+    height: '400px',
+  },
+  formDisplay: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    flexDirection: 'column',
+    rowGap: '10px',
+    margin: '10px',
+  },
+  centerInfo: {
+    textAlign: 'center',
+    marginTop: '10px',
+    marginBottom: '10px',
   },
 });
+
 export const TerminalComponent = () => {
-  const fitAddon = new FitAddon();
-  const terminal = new Terminal();
-  terminal.loadAddon(fitAddon);
-  fitAddon.fit();
+  const [websocketRunning, setWebsocketRunning] = React.useState(false);
+  const [loading, setLoading] = React.useState(false);
+  const [token, setToken] = React.useState('');
+  const [cluster, setCluster] = React.useState('');
```

Comment:
Any reason for these to be 2 separate states if they are always updated together?

Reply:
Good point :+1: I have changed it to signle useState and I have created interface that has both token nad cluster stored in it.